configdialog.cpp

```diff
@@ -1530,3 +1547,128 @@ LRESULT CALLBACK AddPredefinedPaletteDialog(HWND dialog, UINT message, WPARAM wp
 
 	return FALSE;
 }
+
+LRESULT CALLBACK ScreenSaverPatternSelectorDialog(HWND dialog, UINT message, WPARAM wparam, LPARAM lparam) {
+	static std::vector<std::pair<PatternName, std::wstring>> disabled_patterns = { };
+	static std::vector<std::pair<PatternName, std::wstring>> enabled_patterns = { };
```

Comment:
I'm not sure it makes much sense to have both a disabled patterns list AND an enabled patterns list. From what I can tell, this is forcing you to do a lot of extra bookkeeping to make sure these two lists stay in sync - this is a problem because it's complicated and noisy and what happens if you get it wrong? All hell will surely break loose if a pattern shows up in _both or neither_ the disabled set or the enabled set,

The set of enabled patterns can be trivially computed from the set of disabled patterns - it's just the set of all patterns minus the set of disabled patterns (well, this is C++ so it's not trivially implementable, but the idea is trivial). So, whenever someone needs to know the enabled patterns, I think they should just compute that themselves rather than relying on this code to keep track of it. 

There is the wrinkle of course of the Random pattern, which really doesn't belong in either of these because it cannot be disabled or enabled. I'm not really sure what the best approach is here, but it might make sense to actually remove the Random variant from the PatternName enum entirely, because that's not really what it is - conceptually, the user is choosing _either_ the name of a pattern to start with **OR** the special option Random. In this way "the set of all pattern names" is different than "the set of all things the user can choose as a starting pattern", even though in the code they are both modeled by PatternName. These two things should probably be different types!